src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -432,13 +437,13 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 		return cursorYAbsolute > lastCommandYAbsolute;
 	}
 
-	private _cursorLineLooksLikeWindowsPrompt(): boolean {
+	private _getWindowsPrompt(): string | undefined {
 		const line = this._terminal.buffer.active.getLine(this._terminal.buffer.active.baseY + this._terminal.buffer.active.cursorY);
 		if (!line) {
-			return false;
+			return;
 		}
 		// TODO: fine tune prompt regex to accomodate for unique configurtions.
-		return line.translateToString(true)?.match(/^(PS.+>)|([A-Z]:\\.*>)/) !== null;
+		return line.translateToString(true)?.match(/^(?<prompt>(?:PS.+>)|(?:[A-Z]:\\.*>))/)?.groups?.prompt;
```

Comment:
Instead of adding 1 to length, we should change the pwsh part of this to have a space at the end (cmd doesn't have a space at end)